lib/collection/src/shards/shard_holder.rs

```diff
@@ -282,6 +282,20 @@ impl ShardHolder {
         })?)
     }
 
+    /// The count of incoming and outgoing shard transfers on the given peer
+    ///
+    /// This only includes shard transfers that are in consensus for the current collection. A
+    /// shard transfer that has just been proposed may not be included yet.
+    pub fn count_shard_transfer_io(&self, peer_id: &PeerId) -> (usize, usize) {
+        self.shard_transfers
+            .read()
+            .iter()
+            .map(|transfer| (transfer.to == *peer_id, transfer.from == *peer_id))
+            .fold((0, 0), |(i, o), (is_outgoing, is_incoming)| {
+                (i + is_outgoing as usize, o + is_incoming as usize)
+            })
+    }
```

Comment:
```suggestion
    pub fn count_shard_transfer_io(&self, peer_id: &PeerId) -> (usize, usize) {
        let mut outgoing = 0;
        let mut incoming = 0;
        
        for transfer in &self.shard_transfers {
            if transfer.to == *peer_id {
                outgoing += 1;
            }
        
            if transfer.from == *peer_id {
                incoming += 1;
            }
        }
        
        (outgoing, incoming)
    }
```

Reply:
Yours is the wrong way around. `.to` means it'll be _incoming_ on that node, while `.from` means it'll be _outgoing_ on that node.

I mixed up the variable names accidentally in my version, because I flipped the ordering around some commit back. But the result was the same.